os/se/ameba/security_ameba_wrapper.c

```diff
@@ -0,0 +1,1941 @@
+* BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+* OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+* AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+* LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+* ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+* POSSIBILITY OF SUCH DAMAGE.
+*
+***************************************************************************************************/
+
+#include <tinyara/config.h>
+#include <tinyara/seclink_drv.h>
+#include <tinyara/security_hal.h>
+
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include <rom_ssl_ram_map.h>
+
+#define AWRAP_TAG "[AMEBA_WRAPPER]"
+
+#define AWRAP_ENTER                                                                 \
+	do {                                                                             \
+		sedbg(AWRAP_TAG"[INFO] %s %s:%d\n", __FUNCTION__, __FILE__, __LINE__);       \
+	} while (0)
+
+#define HAL_MAX_RANDOM_SIZE 256
+#define HAL_MAX_BUF_SIZE 2048
+#define HAL_MAX_SHARED_SECRET_LEN 256
+#define KEY_STORAGE_INDEX_MAX 32
+
+#define AES_128_KEY_SIZE 16
+#define AES_192_KEY_SIZE 24
+#define AES_256_KEY_SIZE 32
+#define MBEDTLS_HASH_SHA224 1
+#define MBEDTLS_HASH_SHA256 0
+#define MBEDTLS_HASH_SHA384 1
+#define MBEDTLS_HASH_SHA512 0
+
+/* Factory Key and Cert define */
+#define FACTORYKEY_RTL_DEVICE (0x00010120)
+#define FACTORYKEY_RTL_CERT (0x00010122)
+
+#define HAL_COPY_DATA(in, out, len)                     \
+	do {                                                \
+		if (in->data == NULL) {                         \
+			return HAL_INVALID_ARGS;                    \
+		}                                               \
+		memcpy(in->data, out, len);                     \
+		in->data_len = len;                             \
+	} while (0)
+
+#define HAL_COPY_PRIV_DATA(in, out, len)                \
+	do {                                                \
+		if (in->priv == NULL) {                         \
+			return HAL_INVALID_ARGS;                    \
+		}                                               \
+		memcpy(in->priv, out, len);                     \
+		in->priv_len = len;                             \
+	} while (0)
+
+/* ROM mbedtls define */
+#define MBEDTLS_ECP_MAX_BITS    521     /**< Maximum bit size of groups */
+#define MBEDTLS_ECP_MAX_BYTES   ( ( MBEDTLS_ECP_MAX_BITS + 7 ) / 8 )
+#define MBEDTLS_ECDSA_MAX_LEN   ( 3 + 2 * ( 3 + MBEDTLS_ECP_MAX_BYTES ) )
+
+#define MBEDTLS_RSA_PUBLIC      0       /**< Request private key operation. */
+#define MBEDTLS_RSA_PRIVATE     1       /**< Request public key operation. */
+#define MBEDTLS_RSA_PKCS_V15    0       /**< Use PKCS-1 v1.5 encoding. */
+#define MBEDTLS_RSA_PKCS_V21    1       /**< Use PKCS-1 v2.1 encoding. */
+
+typedef enum {
+	MBEDTLS_ECP_DP_NONE = 0,
+	MBEDTLS_ECP_DP_SECP192R1,           /*!< 192-bits NIST curve  */
```

Comment:
duplicated definition to ecp.h which is defined at mbedtls.
please remove these definitions for safety

Reply:
We have remove all mbedtls .h file, these are local define.
Which mean it does not refer to any application mbedtls.

rtl_mbedtls_* is the mbedtls function which is in RTL8721CSM ROM. So we will need these local define to pass in to ROM.
These local define is totally independent to TizenRT application layer mbedtls, as the header file is all remove.

Reply:
we have rename the define on latest commit